etl/steps/data/meadow/abs/2024-08-06/employee_earnings_and_hours_australia.py

```diff
@@ -0,0 +1,85 @@
+"""Load a snapshot and create a meadow dataset."""
+
+import owid.catalog.processing as pr
+from owid.catalog import Table
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap_2006 = paths.load_snapshot("employee_earnings_and_hours_australia_2006.csv")
+    snap_2008 = paths.load_snapshot("employee_earnings_and_hours_australia_2008.csv")
+    snap_2010 = paths.load_snapshot("employee_earnings_and_hours_australia_2010.xls")
+    snap_2012 = paths.load_snapshot("employee_earnings_and_hours_australia_2012.xls")
+    snap_2014 = paths.load_snapshot("employee_earnings_and_hours_australia_2014.xls")
+    snap_2016 = paths.load_snapshot("employee_earnings_and_hours_australia_2016.xls")
+    snap_2018 = paths.load_snapshot("employee_earnings_and_hours_australia_2018.xlsx")
+    snap_2021 = paths.load_snapshot("employee_earnings_and_hours_australia_2021.xlsx")
+    snap_2023 = paths.load_snapshot("employee_earnings_and_hours_australia_2023.xlsx")
+
+    # Load data from snapshot.
+    tb_2006 = snap_2006.read()
+    tb_2008 = snap_2008.read()
+    tb_2010 = snap_2010.read(sheet_name="Table_1", usecols="A,E", skiprows=30, nrows=12)
+    tb_2012 = snap_2012.read(sheet_name="Table_2", usecols="A,J", skiprows=118, nrows=12)
+    tb_2014 = snap_2014.read(sheet_name="Table_2", usecols="A,J", skiprows=5, nrows=12)
+    tb_2016 = snap_2016.read(sheet_name="Table_2", usecols="A,G", skiprows=6, nrows=12)
+    tb_2018 = snap_2018.read(sheet_name="Table_2", usecols="A,G", skiprows=6, nrows=12)
+    tb_2021 = snap_2021.read(sheet_name="Table_2", usecols="A,G", skiprows=6, nrows=12)
+    tb_2023 = snap_2023.read(sheet_name="Table_2", usecols="A,G", skiprows=6, nrows=12)
+
+    #
+    # Process data.
+    #
+    # Format tables. I don't format the 2006 and 2008 tables because they are already in the correct format (they where uploaded manually)
+    tb_2010 = format_tables(tb=tb_2010, year=2010)
+    tb_2012 = format_tables(tb=tb_2012, year=2012)
+    tb_2014 = format_tables(tb=tb_2014, year=2014)
+    tb_2016 = format_tables(tb=tb_2016, year=2016)
+    tb_2018 = format_tables(tb=tb_2018, year=2018)
+    tb_2021 = format_tables(tb=tb_2021, year=2021)
+    tb_2023 = format_tables(tb=tb_2023, year=2023)
+
+    # Merge all tables.
+    tb = pr.concat(
+        [tb_2006, tb_2008, tb_2010, tb_2012, tb_2014, tb_2016, tb_2018, tb_2021, tb_2023],
+        short_name="employee_earnings_and_hours_australia",
+        ignore_index=True,
+    )
```

Comment:
Could define a dictionary with all files and read parameters, and then use a for loop to read all tables & concatenate.

```suggestion

read_params = {
    2006: {"ext": "csv", "params": {}},
    2008: {"ext": "csv", "params": {}},
    2010: {"ext": "xls", "params": {"sheet_name": "Table_1", "usecols": "A,E", "skiprows": 30, "nrows": 12}},
    2012: {"ext": "xls", "params": {"sheet_name": "Table_2", "usecols": "A,J", "skiprows": 118, "nrows": 12}},
    2014: {"ext": "xls", "params": {"sheet_name": "Table_2", "usecols": "A,J", "skiprows": 5, "nrows": 12}},
    2016: {"ext": "xls", "params": {"sheet_name": "Table_2", "usecols": "A,G", "skiprows": 6, "nrows": 12}},
    2018: {"ext": "xls", "params": {"sheet_name": "Table_2", "usecols": "A,G", "skiprows": 6, "nrows": 12}},
    2021: {"ext": "xlsx", "params": {"sheet_name": "Table_2", "usecols": "A,G", "skiprows": 6, "nrows": 12}},
    2023: {"ext": "xlsx", "params": {"sheet_name": "Table_2", "usecols": "A,G", "skiprows": 6, "nrows": 12}},
}


def run(dest_dir: str) -> None:
    #
    # Load & process inputs.
    #
    # Retrieve snapshot.
    tbs = []
    for year, values in read_params.items():
        params = values["params"]
        ext = values["ext"]

        # Load snapshot
        snap = paths.load_snapshot(f"employee_earnings_and_hours_australia_{year}.{ext}")
        # Read snapshot as table
        tb = snap.read(**params)
        # Process table if applciable
        if year not in {2006, 2008}:
            tb = format_tables(tb, year=year)
        # Append to list of tables
        tbs.append(tb)

    # Merge all tables.
    tb = pr.concat(
        tbs,
        short_name="employee_earnings_and_hours_australia",
        ignore_index=True,
    )
```

Reply:
Thank you! I did something similar in the snapshot and I should have done it here as well.